### PR TITLE
Spiderlings ain't no bitches

### PR DESCRIPTION
**Pull request by Zar000**
## About The Pull Request

Spiderlings will now avenge widow when she dies, pursuing any nearby marines and mauling them. This only lasts 10 seconds or untill they die from other causes.
## Why It's Good For The Game

Currently spiderlings just kinda.. look at their dead mother.
Unless widow got gibbed, in which case they would just run around stupidly.
Now however, they will attack any marine in sight, if there are none, they get brazil'd.
## Changelog
:cl:
add: widow spiderlings will avenge widow
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm [hunk 1]
```diff
@@ -78,3 +79,21 @@
 		UnregisterSignal(mob_parent, COMSIG_STATE_MAINTAINED_DISTANCE)
 		UnregisterSignal(atom_to_walk_to, list(COMSIG_MOB_DEATH, COMSIG_PARENT_QDELETING))
 	return ..()
+
+/// This happens when the spiderlings mother dies, they move faster and will attack any nearby marines
+/datum/ai_behavior/spiderling/proc/spiderling_rage()
+	var/mob/living/carbon/xenomorph/spiderling/x = mob_parent
+	var/list/mob/living/carbon/human/possible_victims = list()
+	for(var/mob/living/carbon/human/victim in view(SPIDERLING_RAGE_RANGE, x))
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm:87]: use cheap_get_humans_near

File: code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm [hunk 2]
```diff
@@ -78,3 +79,21 @@
...
+		if(get_dist(x, victim) > SPIDERLING_RAGE_RANGE)
+			continue
+		if(victim.stat == DEAD)
+			continue
+		possible_victims += victim
+	if(isnull(possible_victims))
+		qdel(x)
+		return
+	x.emote("roar")
+	x.next_move_slowdown -= SPIDERLING_RAGE_SPEED
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm:97]: what is the point of that

File: code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm [hunk 3]
```diff
@@ -78,3 +79,21 @@
...
+/datum/ai_behavior/spiderling/proc/spiderling_rage()
+	var/mob/living/carbon/xenomorph/spiderling/x = mob_parent
+	var/list/mob/living/carbon/human/possible_victims = list()
+	for(var/mob/living/carbon/human/victim in view(SPIDERLING_RAGE_RANGE, x))
+		if(get_dist(x, victim) > SPIDERLING_RAGE_RANGE)
+			continue
+		if(victim.stat == DEAD)
+			continue
+		possible_victims += victim
+	if(isnull(possible_victims))
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm:93]: it's not null, you initialized it. Check for size

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Review by Lumipharon — APPROVED** [review]
Looks cool, although I note that the pick() means the spiderlings could ignore like 10 ungas and try rush the dude behind them.

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**